src/hooks/useReplaceableText.ts

```diff
@@ -1,20 +1,30 @@
 import { localisationService, authService } from '@oneblink/apps'
-import * as React from 'react'
+import { useMemo, useCallback } from 'react'
 import useFormSubmissionModel from './useFormSubmissionModelContext'
 import { useRepeatableSetIndexText } from '../form-elements/FormElementRepeatableSet'
 import useTaskContext from './useTaskContext'
+import useReplaceInjectablesOverrides from './useReplaceInjectablesOverrides'
 
 export default function useReplaceableText(text: string) {
+  const replaceInjectablesOverrides = useReplaceInjectablesOverrides()
   const textWithIndex = useRepeatableSetIndexText(text)
   const { formSubmissionModel, elements } = useFormSubmissionModel()
   const { task, taskGroup, taskGroupInstance } = useTaskContext()
-  return React.useMemo(() => {
+
+  const getUserProfile = useCallback(() => {
+    if (replaceInjectablesOverrides?.getUserProfile) {
+      return replaceInjectablesOverrides.getUserProfile()
+    }
+    return authService.getUserProfile()
```

Comment:
Can we swap this to user profile from `useAuth()` please? This will not re-render if user profile changes at the moment from `authService`.